bedrock/products/views.py

```diff
@@ -96,3 +105,135 @@ def vpn_invite_waitlist(request):
         resp = {"success": True}
 
     return JsonResponse(resp)
+
+
+def _build_category_list(entry_list):
+    # Template is expecting this format:
+    # category_list = [
+    #   {"name": "Cat1", "url": "/full/path/to/category"}, ...
+    # ]
+    category_list = []
+    root_url = reverse("products.vpn.resource-center.landing")
+    for entry in entry_list:
+        category = entry.category
+        if category:
+            category_list.append(
+                {
+                    "name": category,
+                    "url": f"{root_url}?{ARTICLE_CATEGORY_LABEL}={quote_plus(category)}",
+                }
+            )
+
+    category_list = sorted(category_list, key=lambda x: x["name"])
+    return category_list
+
+
+def _filter_articles(articles_list, category):
+
+    if not category:
+        return articles_list
+
+    return [article for article in articles_list if article.category == category]
+
+
+def resource_center_landing_view(request):
+
+    ARTICLE_GROUP_SIZE = 6
+
+    active_locales = [
+        "en-US",  # Initially, en-US is the only one available in Contentful
+    ]
+    locale = l10n_utils.get_locale(request)
+
+    template_name = "products/vpn/resource-center/landing.html"
+    ctx = {
+        "active_locales": active_locales,
+    }
+
+    resource_articles = ContentfulEntry.objects.get_entries_by_type(
+        locale=locale,
+        classification=CONTENT_CLASSIFICATION_VPN,
+        content_type=CONTENT_TYPE_PAGE_RESOURCE_CENTER,
+    )
+    category_list = _build_category_list(resource_articles)
+    selected_category = unquote_plus(request.GET.get(ARTICLE_CATEGORY_LABEL, ""))
+
+    _filtered_articles = _filter_articles(
+        resource_articles,
+        category=selected_category,
+    )
+
+    # The resource_articles are ContentfulEntry objects at the moment, but
+    # we really only need their JSON data from here on
+    _filtered_article_data = [x.data for x in _filtered_articles]
+
+    first_article_group, second_article_group = (
+        _filtered_article_data[:ARTICLE_GROUP_SIZE],
+        _filtered_article_data[ARTICLE_GROUP_SIZE:],
+    )
+
+    ctx.update(
+        {
+            "category_list": category_list,
+            "selected_category": selected_category,
```

Comment:
if we're sending `selected_category`, which we got from user input, to the template, we need to ensure it's sanitized to avoid XSS possibilities.

Reply:
Good point

- [ ] @stevejalim sanitize

Reply:
@pmac done